### DISCUSSION TITLE
请问有升级数据库的方法吗

### DISCUSSION THREAD

**Discussion by pardont** [dotnetcore/FreeSql#1266] [General]
在应用升级过程中，可能会添加新表或者添加新字段，有没有方便的升级数据库的方法呢？

我的想法是：
读取旧数据库的表和表的字段
和新数据库对比，如果有新表或新字段，就生成更改脚本来升级旧数据库

freesql中有类似的实现吗？
如果没有现成的实现，可以给一些建议吗？

感谢。

**Comment by 2881099** [comment]
freesql 支持实体类与表对比，但是生产环境需要慎重。

建议使用专业的 DB 管理工具做这件事件，人工审核后执行。

例如，navicat 就可以对比两个表的结构。

**Reply by pardont** [reply]
我们是一个c/s应用，每次升级的时候该应用的本地sqlite数据库可能会加字段或者加表。
能告诉我怎么调用吗？我看看实现的思路，测试一下

**Reply by 2881099** [reply]
看下 CodeFirst 文档，里面有API介绍，主要两个：

把实体类与表对比，返回 DDL：string fsql.CodeFirst.GetComparisonDDLStatements(typeof(T), tablename)

把实体类与表对比，执行迁移：void fsql.CodeFirst.SyncStructure(typeof(T), tablename)

执行迁移，记得备份数据库。

**Reply by pardont** [reply]
收到，万分感谢
